src/ldp/http/metadata/BasicMetadataHandler.ts

```diff
@@ -0,0 +1,29 @@
+import type { HttpRequest } from '../../../server/HttpRequest';
+import { RepresentationMetadata } from '../../representation/RepresentationMetadata';
+import { MetadataHandler } from './MetadataHandler';
+import type { MetadataParser } from './MetadataParser';
+
+/**
+ * MetadataHandler that lets each of its MetadataParsers add metadata based on the HttpRequest.
+ */
+export class BasicMetadataHandler extends MetadataHandler {
```

Comment:
Only thing I'm thinking is this might be confusing with the CompositeAsyncHandler we already have since this is currently only needed here for the metadata. Unless the question is why we're not using that one, for which I refer to my main comment in the PR above.

Reply:
OK, got it. Yeah, I think there's two types of composites. The one that picks the first, and the one that picks them all?
Not sure if we want to make that generic, but sounds nice. Nothing is specific to metadata here.

Reply:
I think the one thing stopping me from doing it is that it feels inefficient to me to have them throw errors if they can't do anything with the input headers since there might be a lot of those handlers in the future. But that might be too much of a micro optimization.